pykdtree/kdtree.pyx

```diff
@@ -162,7 +162,7 @@ cdef class KDTree:
 
         # Get tree info
         self.n = <uint64_t>data_pts.shape[0]
-        self._use_int32_t = self.n < UINT32_MAX
+        self._use_int32_t = self.n * <uint64_t>data_pts.shape[1] < UINT32_MAX
```

Comment:
Why is the cast to uint64 needed? I suppose it is a Python int otherwise?

Reply:
I don't think it's needed, I might be getting a bit paranoid about integer overflows!